renku/core/dataset/dataset_add.py

```diff
@@ -155,6 +159,21 @@ def _download_files(
     **kwargs,
 ) -> List["DatasetAddMetadata"]:
     """Process file URLs for adding to a dataset."""
+    if dataset.storage and dataset.storage.lower().startswith("s3://"):
```

Comment:
I think `and dataset.storage.lower().startswith("s3://")` is not needed.

Reply:
I kinda changed/removed this whole section. I did this by requiring that the scheme of `dataset.storage` matches the scheme of the uri being added.